flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommittableCollectorSerializerTest.java

```diff
@@ -72,38 +73,88 @@ void testCommittableCollectorV1SerDe() throws IOException {
-                                .collect(Collectors.toList()))
-                .containsExactly(2);
+        assertThat(copy.getNumberOfSubtasks()).isEqualTo(numberOfSubtasks);
+
+        // assert original CommittableCollector
+        assertCommittableCollector(
+                "Original CommittableCollector", subtaskId, numberOfSubtasks, committableCollector);
+
+        // assert deserialized CommittableCollector
+        assertCommittableCollector(
+                "Deserialized CommittableCollector", subtaskId, numberOfSubtasks, copy);
+    }
+
+    private void assertCommittableCollector(
+            String assertMessageHeading,
+            int subtaskId,
+            int numberOfSubtasks,
+            CommittableCollector<Integer> committableCollector) {
+
+        assertAll(
+                assertMessageHeading,
+                () -> {
+                    final Collection<CheckpointCommittableManagerImpl<Integer>>
+                            checkpointCommittables =
+                                    committableCollector.getCheckpointCommittables();
+                    assertThat(checkpointCommittables).hasSize(2);
+
+                    final Iterator<CheckpointCommittableManagerImpl<Integer>> committablesIterator =
+                            checkpointCommittables.iterator();
+                    final CheckpointCommittableManagerImpl<Integer> checkpointCommittableManager1 =
+                            committablesIterator.next();
+                    final SubtaskCommittableManager<Integer> subtaskCommittableManagerCheckpoint1 =
+                            checkpointCommittableManager1.getSubtaskCommittableManager(subtaskId);
+                    assertThat(checkpointCommittableManager1.getSummary().getSubtaskId())
+                            .isEqualTo(subtaskId);
+                    assertThat(checkpointCommittableManager1.getSummary().getNumberOfSubtasks())
+                            .isEqualTo(numberOfSubtasks);
+                    assertThat(
+                                    subtaskCommittableManagerCheckpoint1
+                                            .getPendingRequests()
+                                            .map(CommitRequestImpl::getCommittable)
+                                            .collect(Collectors.toList()))
+                            .containsExactly(1);
+                    assertThat(subtaskCommittableManagerCheckpoint1.getSubtaskId())
+                            .isEqualTo(subtaskId);
+
+                    final CheckpointCommittableManagerImpl<Integer> checkpointCommittableManager2 =
+                            committablesIterator.next();
+                    final SubtaskCommittableManager<Integer> subtaskCommittableManagerCheckpoint2 =
+                            checkpointCommittableManager2.getSubtaskCommittableManager(subtaskId);
+                    assertThat(checkpointCommittableManager2.getSummary().getSubtaskId())
```

Comment:
Nit: You can use `SinkV2Assertions.assertThat` to simplify the assertions here or extend it also to cover the subtaskId.

Reply:
Fixed, plz take a look.